src/commands/azureServiceOperators/helpers/azureservicehelper.ts

```diff
@@ -0,0 +1,287 @@
+        return undefined;
+    }
+}
+
+export async function installOperatorSettings(
+    kubectl: k8s.APIAvailable<any>,
+    operatorSettingInfo: OperatorSettings,
+    clusterKubeConfig: string
+): Promise<k8s.KubectlV1.ShellResult | undefined> {
+    try {
+        const extensionPath = getExtensionPath();
+        const yamlPathOnDisk = vscode.Uri.file(path.join(extensionPath!, 'resources', 'yaml', 'azureoperatorsettings.yaml'));
+
+        const azureoperatorsettings = fs.readFileSync(yamlPathOnDisk.fsPath, 'utf8');
+        azureoperatorsettings
+            .replace("<TENANT_ID>", operatorSettingInfo.tenantId)
+            .replace("<SUB_ID>", operatorSettingInfo.subId)
+            .replace("<APP_ID>", operatorSettingInfo.appId)
+            .replace("<CLIENT_SECRET>", operatorSettingInfo.clientSecret)
+            .replace("<ENV_CLOUD>", operatorSettingInfo.cloudEnv);
+
+        const templateYaml = tmp.fileSync({ prefix: "aso-operatorsettings", postfix: `.yaml` });
+        fs.writeFileSync(templateYaml.name, azureoperatorsettings);
+
+        const runCommandResult = await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
+            clusterKubeConfig, "YAML",
+            (f) => kubectl.api.invokeCommand(`apply -f ${templateYaml.name} --kubeconfig="${f}"`));
+
+        return runCommandResult;
+    } catch (e) {
+        vscode.window.showErrorMessage(`Install operator settings had following error: ${e}`);
+        return undefined;
+    }
+}
+
+export function getWebviewContent(
+    clustername: string,
+    aksExtensionPath: string,
+    installationResponse: InstallationResponse
+): string {
+    const stylePathOnDisk = vscode.Uri.file(path.join(aksExtensionPath, 'resources', 'webviews', 'azureserviceoperator', 'azureserviceoperator.css'));
+    const htmlPathOnDisk = vscode.Uri.file(path.join(aksExtensionPath, 'resources', 'webviews', 'azureserviceoperator', 'azureserviceoperator.html'));
+    const styleUri = stylePathOnDisk.with({ scheme: 'vscode-resource' });
+    const pathUri = htmlPathOnDisk.with({ scheme: 'vscode-resource' });
+
+    const htmldata = fs.readFileSync(pathUri.fsPath, 'utf8').toString();
+    const certManagerResult = installationResponse.installCertManagerResult;
+    const certManagerSatatusResult = installationResponse.checkCertManagerRolloutStatusResult;
+    const issuerCertResult = installationResponse.installIssuerCertResult;
+    const olmCrdResult = installationResponse.installOlmCrdResult;
+    const olmResult = installationResponse.installOlmResult;
+    const operatorResult = installationResponse.installOperatorSettingsResult;
+    const operatorSettingsResult = installationResponse.installOperatorResult;
+    const getOperatorsPodResult = installationResponse.getOperatorsPodResult;
+
+    let exitCode = 0;
+
+    // If total of outputCode is more than 0 then obviously something failed and display the html error, with std ouput.
```

Comment:
Be very careful! It is possible for an exit code to be interpreted as a negative number I believe... Consider instead creating an array of results and writing `allSucceeded = results.all(sr => !!sr && sr.code == 0)`.  It may feel less efficient but it is safer and more intention-revealing.

Reply:
Oh. your function only does anything if the input exit code is 0. So once you have a non-zero you discard all others anyway, so the comment is out of date.

I still prefer the `.all(... === 0)` though. *grin*